homeassistant/components/sonos/favorites.py

```diff
@@ -0,0 +1,79 @@
+"""Class representing Sonos favorites."""
+from __future__ import annotations
+
+from collections.abc import Iterator
+import datetime
+import logging
+from typing import Callable
+
+from pysonos.core import SoCo
+from pysonos.data_structures import DidlFavorite
+from pysonos.events_base import Event as SonosEvent
+from pysonos.exceptions import SoCoException
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import dispatcher_send
+
+from .const import SONOS_HOUSEHOLD_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SonosFavorites:
+    """Storage class for Sonos favorites."""
+
+    def __init__(self, hass: HomeAssistant, soco: SoCo) -> None:
+        """Initialize the data."""
+        self.hass = hass
+        self.soco = soco
+        self._favorites: list[DidlFavorite] = []
+        self._event_version: str | None = None
+        self._next_update: Callable | None = None
+
+    def __iter__(self) -> Iterator:
+        """Return an iterator for the known favorites."""
+        return iter(self._favorites)
```

Comment:
Is there any risk of this changing during iteration?

Reply:
It would be extremely rare, but I suppose there is a possibility of someone hitting that eventually.

Reply:
Can we return a copy of the list and iterate that instead ?

Reply:
Probably safer to return a copy. Will adjust.